test/tools/router_check/router_check.cc

```diff
@@ -1,12 +1,46 @@
 // NOLINT(namespace-envoy)
 #include <iostream>
+#include <memory>
 #include <string>
+#include <vector>
 
+#include "source/common/common/fmt.h"
 #include "source/common/common/thread.h"
+#include "source/common/protobuf/utility.h"
 #include "source/exe/platform_impl.h"
 
 #include "test/test_common/test_runtime.h"
 #include "test/tools/router_check/router.h"
+#include "test/tools/router_check/validation.pb.h"
+
+bool hasFailures(
+    const std::vector<envoy::RouterCheckToolSchema::ValidationItemResult>& test_results) {
+  for (const auto& test_result : test_results) {
+    if (!test_result.test_passed()) {
+      return true;
+    }
+  }
+  return false;
+}
+
+void writeOutput(const envoy::RouterCheckToolSchema::ValidationResult& result,
```

Comment:
Should be replace by `TestEnvironment::writeStringToFileForTest()` if possible.

Reply:
fixed